### PR TITLE
Fix 557 - invalid preprocessor for id of "reg"

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Changes
RC - fixed so that ids of reg and rsv are bible types.  Added unit test case.

### CODE REVIEW

File: resource_container/ResourceContainer.py [hunk 1]
```diff
@@ -334,7 +334,12 @@ def identifier(self):
         if 'identifier' in self.resource and self.resource['identifier']:
             return self.resource['identifier']
         if 'id' in self.resource and self.resource['id']:
-            return self.resource['id']
+            id = self.resource['id']
```
Comment (richmahn) [inline resource_container/ResourceContainer.py]: My only concern would be if this repo had a v0.2 manifest.yaml file the first if would be what passes. I'm actually thinking we should just add those resource types to the possibe Bible types, here https://github.com/unfoldingWord-dev/tx-manager-lambda/blob/develop/functions/convert_usfm2html/module.json#L5 and here https://github.com/unfoldingWord-dev/tx-manager/blob/develop/converters/usfm2html_converter.py#L15

**Review by richmahn — CHANGES_REQUESTED** [review]
I suggest that we change it in the actual place these resource types are accepted.

File: resource_container/ResourceContainer.py (hunk 1 above)
Comment (PhotoNomad0) [reply resource_container/ResourceContainer.py]: @richmahn - OK, this is going back to my original approach.  I just misunderstood your suggested changes.

**Review by richmahn — APPROVED** [review]

**Review by richmahn — APPROVED** [review]
:+1: LGTM

**Pull request merged**